### PR TITLE
CI: set up automated updates for github actions

### PR DESCRIPTION
**Pull request by hollandjg**

### CODE REVIEW

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
Thanks for this! Just two obsevations.

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -4,13 +4,12 @@ on:
   push:
     branches: [ "main" ]
   pull_request:
```
[the remark this answers is not in the document]
Comment (hollandjg) [reply .github/workflows/linter.yml:6]: I'm not sure which line you mean. I think we might want both checks running on push to main and on all pull requests. What do you think?

[the remark this answers is not in the document]
Comment (cpaniaguam) [reply .github/workflows/linter.yml:6]: I was thinking L6 was not needed as the line below it was deleted. I was a bit confused by the scoping of the `push` trigger to the main branch only but the `pull_request` one to all branches. Is that what you intended?

[the remark this answers is not in the document]
Comment (cpaniaguam) [reply .github/workflows/linter.yml:6]: > I think we might want both checks running on push to main and on all pull requests. What do you think?
I think that is reasonable!

**Review by cpaniaguam — APPROVED** [review]

File: .github/workflows/linter.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hollandjg) [reply .github/workflows/linter.yml:6]: Oh, I understand better now. With `pull_request:` we make sure that the linter checks get run on every PR, not just the ones targeting main. The motivation is that I've got a chain of PRs, but right now only the one closest to `main` gets the status checks.

[the remark this answers is not in the document]
Comment (hollandjg) [reply .github/workflows/linter.yml:6]: (I'm not wedded to having them run on push to main – I think that's unnecessary if we always do status checks and always use PRs.)

**Pull request merged**